backend/core/rule/tree.py

```diff
@@ -120,7 +234,31 @@ def find(self, name):
 
 class TreeNode(object):
     def __init__(self, data, properties, metadata, func=None, namespace=None, *args):
+        """Constructor of TreeNode
+
+        Args:
+            data: data of current node. more additional explain is in below
+            properties: dict of property. will be deconstructed
+            metadata: data of any other information needed evaluating tree
+            func: evaluator function (closure)
+            namespace: name which user actually see
+            *args: any other additional arguments
+        """
+
+        # self.data can contain various type of data.
+        # If type of self.data is `None`, it means it is not
+        # yet evaluated. Maybe it is not leaf node, which
+        # means for being evaluate, all children node
+        # should be evaluated.
```

Comment:
request does not have identical meaning with mine. But, I found one grammatic mistake. I'll fix this.